addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -180,3 +201,72 @@ def _timesheet_postprocess_values(self, values):
                     'amount': amount_converted,
                 })
         return result
+
+    # ----------------------------------------------------
+    # Timer Methods
+    # ----------------------------------------------------
+    def _compute_display_timer(self):
+        """ Check if the encoding unit is 'Hours',
+
+            if yes then the timer button is displayed
+        """
+        for record in self:
+            if record.encoding_uom_id.id == self.env.ref('uom.product_uom_hour').id:
+                record.display_timer = True
+            else:
+                record.display_timer = False
+
+    def action_timer_start(self):
+        """ Action start the timer of the current timesheet.
+
+            Start timer and search if another timer hasn't been launched.
+            If yes, then stop the timer before launch this timer.
+        """
+        self.ensure_one()
+        if not self.timer_start and self.display_timer:
+            self._search_timer_start()
+            self.write({
+                'timer_start': fields.Datetime.now(),
+            })
+
+    def _search_timer_start(self):
+        """ Search if a timesheet has a timer activated and stop the timer.
+
+            Check if a timer is activated for another timesheet
+            if yes, then update unit_amount field and stop timer,
+            otherwise, do nothing.
+        """
+        record = self.env['account.analytic.line'].search([
+            ('timer_start', '!=', False),
+            ('user_id', '=', self.env.uid)
+        ])
+
+        if record:
+            record.action_timer_stop()
+
+    def action_timer_stop(self):
+        """ Action stop the timer of the current timesheet.
+
+            When the timer must be stopped, we must calculate the new
+            unit_amount based on the timer and the previous value of
+            unit_amount for the current timesheet.
+        """
+        if self.timer_start and self.display_timer:
+            precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+            minutes_spent = self._compute_minutes_spent()
+
+            if float_is_zero(self.unit_amount, precision_digits=precision) and float_is_zero(minutes_spent, precision_digits=precision):
+                # Check if unit_amount and minutes_spent is not 0,
+                # if yes, we set minutes_spent to 1.
```

Comment:
Why?

Reply:
Looks like a quick hack nope ?

Reply:
I change this condition because it doesn't work the way I want it to. The idea is to check if the unit_amount is equal to 0 and minutes_spent < 1, then we have a timesheet with unit_amount field set on 0. And afterwards, we can invoice this timesheet, it's useless. To not have this, I initialize the minutes_spent to 1 (if the condition is True) and then we don't have the timesheet with unit_amount to 0 when we stop the timer.

Reply:
By looking at this a second time, the behavior is really strange.
For a brand new timesheet, 10s counts as a full minute.
But after, 10s counts as 0 minute.
:swimmer:
I would ask AJU again if she really wants this.